src/crypto/mod.rs

```diff
@@ -7,7 +7,8 @@ use std::io;
 use crate::util::*;
 
 use rand_chacha::ChaCha12Rng;
-use rand_core::{self, RngCore, SeedableRng};
+use rand_chacha::rand_core::{RngCore as _, SeedableRng};
```

Comment:
To explain: `rand_chacha` 0.3 uses `rand_core` 0.6, which we don't currently hold. So, we use its re-exports here to make it happy. I'll try to figure out something nicer.

Reply:
(We don't hold `rand_core` 0.6 or 0.7 because `rand` *itself* depends on 0.5, and we use its traits...)